libnmstate/netapplier.py

```diff
@@ -164,6 +165,20 @@ def sanitize_ethernet_state(ifaces_desired_state, ifaces_current_state):
     return ifaces_desired_state
 
 
+def sanitize_dhcp_state(ifaces_desired_state):
+    """
+    If DHCP enabled, if desired state does not contain any IP address,
+    the 'address' value should be keep as empty list instead of copying from
+    current state.
+    """
+    for ifname, iface_state in six.viewitems(ifaces_desired_state):
+        for family in ('ipv4', 'ipv6'):
+            ip = iface_state.get(family, {})
+            if ip.get('enabled'):
```

Comment:
Did you mean `dhcp` and `autoconf`?
You should not set it for non `dynamic settings.